platform_setup.sh

```diff
@@ -1,62 +1,60 @@
-echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> ~/.zshrc
-/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
-
-# Install Heroku CLI & Heroku Multiple Accounts Plugin
-brew install heroku/brew/heroku
+#!/usr/bin/env bash
+
+function append_config() {
+  if [ ! -f $1 ]
+  then
+    return
+  fi
+  if grep -q "$2" "$1"
+  then
+    return
+  fi
+  echo "Appending to $1: $2"
+  echo >> "$1"
+  echo "# Added by platform_setup.sh" >> "$1"
+  echo "$2" >> "$1"
+}
+
+function append_file() {
+  append_config "$HOME/.zshrc" "$1"
+  append_config "$HOME/.bash_profile" "$1"
+}
+
+if pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | grep -q "version"
+then
+  echo "XCode CLI Tools already installed!"
+else
+  echo "Installing XCode CLI Tools..."
+  sudo xcode-select --install
+fi
+
+if command -v brew | grep -q "brew"
+then
+  echo "Homebrew already installed!"
+else
+  echo "Installing Homebrew..."
+  curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh -o brew.sh
+  source brew.sh
+fi
+
+echo "Creating NVM's working directory if it doesn't exist..."
+mkdir -p ~/.nvm
+
+echo "Load Homebrew when the shell starts..."
+append_file 'eval "$(/opt/homebrew/bin/brew shellenv)"'
+eval "$(/opt/homebrew/bin/brew shellenv)"
+
+echo "Installing Brewfile..."
+brew bundle
+
+echo "Load NVM when the shell starts..."
+cp -np ./.start_nvm ~
+append_file "source ~/.start_nvm"
+source ./.start_nvm
+
+echo "Add Heroku Multiple Accounts Plugin..."
 heroku plugins:install heroku-accounts
 
-# Install PostgreSQL - Database
-brew install postgresql
-brew services start postgresql
-
-# Install Redis - In-memory data structure store
-brew install redis
-brew services start redis
-
-# Install NVM - Node version manager allows us to easily change node versions
-brew install nvm
-
-# You should create NVM's working directory if it doesn't exist:
-[ -d ~/.nvm ] || mkdir ~/.nvm
-
-# Add the following to ~/.zshrc or your desired shell configuration file:
-echo 'export NVM_DIR="$HOME/.nvm"' >> ~/.zshrc
-# This loads nvm
-echo '  [ -s "/usr/local/opt/nvm/nvm.sh" ] && \. "/usr/local/opt/nvm/nvm.sh"' >> ~/.zshrc
- # This loads nvm bash_completion
-echo '  [ -s "/usr/local/opt/nvm/etc/bash_completion.d/nvm" ] && \. "/usr/local/opt/nvm/etc/bash_completion.d/nvm"' >> ~/.zshrc
-
-# re-source your terminal
-source ~/.zshrc
-
-# Install Node using NVM
-# Install node version
-nvm install 16.4.1
-# Set node version
-nvm use 16.4.1
-# Set as system default
-nvm alias default 16.4.1
-
-# Install puma-dev - Development server with HTTPS support
-brew install puma/puma/puma-dev
-
-# Install Yarn - Package manager for Node apps
-brew install yarn
-
-# Install and configure Pry for a better rails console experience
-gem install pry
-# Add the following lines to the file ~/.pryrc
-echo 'Pry.config.pager = false' >> ~/.pryrc
-echo 'require "awesome_print"' >> ~/.pryrc
-echo 'AwesomePrint.pry!' >> ~/.pryrc
```

Comment:
Do you think we should install these as part of the valiant repo `setup_dev.sh` script? Maybe even add it as a option rather than force devs to use it.

Reply:
Definitely happy with removing / making optional over the forced install. 
I'll be using it every single time, so personally I'd love it as a one and done.

Reply:
I'll change this to copy `.pryrc` only if it doesn't already exist for the user.